app/models/illumina_htp/mx_tube_purpose.rb

```diff
@@ -48,7 +48,7 @@ def source_plate_scope(tube)
       .joins(wells: :requests)
       .where(
         requests: {
-          target_asset_id: tube.id,
+          target_asset_id: tube.receptacle.id,
```

Comment:
why the change here? (or why it worked before)?

Reply:
I'm pretty sure it's a bug and it didn't work before. I mean, it wouldn't have thrown an exception, but it would have returned the wrong data. The target_asset field is meant to be a receptacle id, not a labware id.